_source/_posts/2020-10-12-build-modern-api-using-fastify.md

```diff
@@ -0,0 +1,247 @@
+---
+layout: blog_post
+title: "Build a Modern API using Fastify and Node.js"
+author: nickolas-fisher
+by: contractor
+communities: [javascript]
+description: "Fastify is fast! Learn to build a modern Node.js API using Fastify and JSON web tokens!"
+tags: [javascript,nodejs,fastify,api,jwt]
+tweets:
+- "Fastify is fast! In this tutorial, you'll learn to build a modern Node.js API using Fastify! #nodejs #fastify @fastifyjs"
+- "Learn how to build a modern Node.js API using Fastify! #nodejs #fastify @fastifyjs"
+- "Fastify is designed to be the fastest framework for Node.js. Here's how you can get started building a modern, secure API using Fastify! #nodejs #fastify @fastifyjs"
+image: blog/featured/okta-node-skew.jpg
+type: conversion
+---
+
+Fastify is just as the name implies, fast. Not just in terms of development speed—its low overhead means the server is fast as well. When writing APIs, speed on both sides is paramount. Fastify is a web framework for Node.js that was designed for efficiency. Fastify is fully extensible with hooks, plugins, and decorators. It is schema-based, meaning you can define your response and request objects in your routes and have Fastify do the data validation for you. This feature isn't mandatory, but it is useful. On the request side, it adds easy data validation. On the response side, this means you can shape your data for even lower overhead by not pushing down extraneous data to the client. Fastify is also TypeScript friendly. However, in this tutorial you will use JavaScript.
+
+For this tutorial, you will create a secure API that returns some information regarding employees. To authenticate users, you will use Okta as an authentication server which will produce a JSON Web Token (JWT) after authenticating the user. The client will then send that JWT as part of the request to the server which will handle the validation logic. You will use Okta's JWT verifier to quickly set up the authentication on your server.
+
```

Comment:
Lot of "will" language, which weakens the point of view.

"For this tutorial, you create..." is stronger. Not a deal breaker, though. ;)